### PR TITLE
Refactor JWT signing function

### LINKED ISSUE #116: Break out jwt signing in a separate module
The current codebase have one jwt function in the EGA module and one in the Elixir module, we only need one.

### PR DESCRIPTION
**Pull request by jbygdell**
* Creates unified JWT signing function so the tokens look similar for both EGA and LS-AAI
* Removes old EGA JWT function
* Removes old Elixir JWT function
* Fixes dev environment
* Deprecates `io/ioutil`
* Start HTTP server with timouts
* Update to GO 1.19

Fixes #116

### CODE REVIEW

**Review by norling — APPROVED** [review]
some minor things, otherwise I think it's good! jwt's aren't my strong suit though

**Comment by aaperis** [comment]
Tried to run it with the compose file but `auth` failed with:
`time="2022-11-14T13:57:31Z" level=error msg="State of incoming request (4636d955-f40f-4e17-a139-3b1af81d40b2) does not match with your session's state ()"` . Seems the cookie is not being written properly, but don't know why.

File: config.yaml [hunk 1]
```diff
@@ -8,18 +8,16 @@ elixir:
   redirectUri: "http://oidc:31111/elixir/login"
   revocationUrl: "http://oidc:9090"
   secret: "wHPVQaYXmdDHg"
-  jwtPrivateKey: "keys/sign-rsa-jwt.key"
```
Comment (aaperis) [inline config.yaml:11]: Since these keys are not used anymore, `readme` needs to be updated accordingly.

**Review by aaperis — CHANGES_REQUESTED** [review]
it is definitely much clearer now but some issues need another look :)

File: elixir.go [hunk 2]
```diff
@@ -107,76 +103,6 @@ func authenticateWithOidc(oauth2Config oauth2.Config, provider *oidc.Provider, c
...
-
-	if jwksURL == "" {
-		token, _ = jwt.Parse(idStruct.Token, func(tokenElixir *jwt.Token) (interface{}, error) { return nil, nil })
-	} else {
-		token, err = validateToken(idStruct.Token, jwksURL)
-		if err != nil {
-			return "", "", fmt.Errorf("could not validate raw jwt against pub key, reason: %v", err)
-		}
-		log.WithFields(log.Fields{"authType": "elixir", "user": idStruct.User}).Infof("Token was validated")
-	}
```
Comment (aaperis) [inline elixir.go:127]: With this out, we do not validate the elixir token against the jwk endpoint anymore.

Comment (jbygdell) [reply elixir.go:127]: This is a prefect example of misplaced code, this should be in the OIDC authentication section

**Review by aaperis — APPROVED** [review]

**Review by dbampalikis — APPROVED** [review]

**Pull request merged**